### DISCUSSION TITLE
Ambiguous Onestop IDs

### DISCUSSION THREAD

**Discussion by bamyers99** [transitland/transitland#382] [Q&A] [answered]
Stations and platforms have the same id. Some routes have the same id.
https://www.transit.land/stops/s-9q9hzd44m3-crossmanstation
Station PS_CRSS and platform 64793 have the same coordinates and name so their Onestop IDs are the same.

Raw GTFS data:
PS_CRSS,Crossman Station,PS_CRSS,,37.408841,-122.01118,,,,,1,,America/Los_Angeles,1,
64793,Crossman Station,64793,Eastbound,37.408841,-122.01118,,,,,0,PS_CRSS,America/Los_Angeles,1,

More ambiguous station/platforms:
https://www.transit.land/stops/s-9q9k1m07rz-hamiltonstation
https://www.transit.land/stops/s-9q9k1qjrmh-bascomstation
https://www.transit.land/stops/s-9q9k3c7k1t-racestation
https://www.transit.land/stops/s-9q9k63f7f8-virginiastation
https://www.transit.land/stops/s-9q9k65u90h-sanfernandostation
https://www.transit.land/stops/s-9q9k65r9zx-childrensdiscoverymuseumstation
https://www.transit.land/stops/s-9q9k6nmnpm-japantown~ayerstation

Ambiguous routes:
https://www.transit.land/operators/o-dr5r-path routes 860, 861, 1024 all have the same id https://www.transit.land/routes/r-dr5re-path

**Comment by irees** [comment, marked as answer]
Yes, this is by design. In the Transitland v2 APIs, a Onestop ID can map to more than one GTFS entity. They function essentially as a stable lookup key for the current data. This allows the API end user to decide when and if to merge these entities together.
